src/widgets/main_window/__init__.py

```
@@ -36,29 +38,92 @@ def initUi(self):
+        iwadLayout.setContentsMargins(0, 0, 0, 0)
+        iwadLayout.addWidget(self.iwadInput)
+        iwadLayout.addWidget(self.iwadBrowseButton)
+        self.iwadInputContainer.setLayout(iwadLayout)
         self.pwadListLabel = QLabel("PWAD List:")
         self.pwadList = PWadList()
         self.pwadList.setSizePolicy(
             QSizePolicy.Expanding, QSizePolicy.Maximum)
+        for wad in self.config.get('lastPWads', []):
+            self.pwadList.addWad(wad)
+        self.pwadList.orderChanged.connect(self.saveConfig)
+        self.pwadAddButton = QPushButton('Add...')
```

Ruff (0.12.2)

`QPushButton` may be undefined, or defined from star imports

(F405)